### PR TITLE
TE-44: add modal for MenuItemCard

### PR DESCRIPTION
**Pull request by RomanLavrunov**

### CODE REVIEW

File: components/Modal.tsx [hunk 1]
```diff
@@ -3,29 +3,38 @@ import Typography from '@mui/material/Typography';
...
-          <IconButton disableFocusRipple={true} className="modal-content-title__icon-button" onClick={close}>
-            <CloseIcon />
-          </IconButton>
-        </Typography>
-        <Typography component="div" id="modal-modal-description" sx={{ mt: 2 }}>
+      <Box className={classnames('modal-content', { 'modal-content--constrained': isConstrained }, { 'modal-content__item-card': !isConstrained })}>
+        <IconButton disableFocusRipple={true} className="modal-content-title__icon-button" onClick={close}>
+          <CloseIcon />
+        </IconButton>
+        {modalTitle}
```
Comment (devmc-ee) [inline components/Modal.tsx:36]: What is the benefit of extracting this title into a separate component (or sub-component) in comparison with regular conditional rendering like 
```typescript
{title 
? 
 (
<Typography className="modal-content-title" variant="h5" component="h3">
    {title}
</Typography> 
)
: null }
```

And do we need to hide the entire component, or it is just enough to hide an empty text?
``` typescript
  {title || ''}
```

File: menu/MenuItemCard/MenuItemCard.tsx [hunk 2]
```diff
@@ -12,34 +12,61 @@ import { opacity } from '@cloudinary/url-gen/actions/adjust';
 interface MenuItemCardProps {
   menuItemData: IMenuItem;
   cdn: Cloudinary;
+  isModal?: boolean;
```
Comment (devmc-ee) [inline menu/MenuItemCard/MenuItemCard.tsx:15]: Looks like the logic of 2 different components mixed in a single one,  and such boolean is a bright sign of it.  Should we split simple MenuItemCard from MenuItemCardModal?

**Review by devmc-ee — CHANGES_REQUESTED** [review]

File: next.config.js [hunk 3]
```diff
@@ -6,9 +6,9 @@ module.exports = {
   poweredByHeader: false,
   i18n,
   env: {
-    apiUrl: process.env.NODE_ENV === 'production' ? 'https://api.milicity.eu' : 'http://localhost:5003',
+    apiUrl: process.env.NODE_ENV !== 'production' ? 'https://api.milicity.eu' : 'http://localhost:5003',
```
[the remark this answers is not in the document]
Comment (RomanLavrunov) [reply next.config.js]: I forgot to change it in the first commit. Made a second one to channge this.

File: pages/menu/index.tsx [hunk 4]
```diff
@@ -219,4 +226,9 @@ export const getServerSideProps: GetServerSideProps<MenuPageProps> = async (cont
       ...translations,
     }
   };
-};
\ No newline at end of file
+};
+
```
[the remark this answers is not in the document]
Comment (RomanLavrunov) [reply pages/menu/index.tsx]: Surplus lines deleted

File: package-lock.json [hunk 5]
```diff
@@ -2069,6 +2114,141 @@
         "node": ">= 10"
       }
     },
+    "node_modules/@next/swc-freebsd-x64": {
```
[the remark this answers is not in the document]
Comment (RomanLavrunov) [reply package-lock.json]: Frankly speaking, I have no clue where it came from. I run this prjoect on Mac and haven't installed anything, nor adjusted anything manually there. I updated the the pacakage-lock, now it doesn't have this line of code.

File: components/Modal.tsx (hunk 1 above)
Comment (RomanLavrunov) [reply components/Modal.tsx]: Yep, you correct as usual. Changed the code to what you suggested.

File: menu/MenuItemCard/MenuItemCard.tsx (hunk 2 above)
Comment (RomanLavrunov) [reply menu/MenuItemCard/MenuItemCard.tsx]: I had that feeling in the end, whereas in the beginning it seemed like a good idea.. :)

**Review by devmc-ee — APPROVED** [review]

**Pull request merged**